### PR TITLE
Uniformize download file sizes

### PR DESCRIPTION
**Pull request by FarisMarouane**

### CODE REVIEW

File: downloadSpeed.js [hunk 1]
```diff
@@ -1,75 +1,25 @@
...
-      }
-    });
+  promiseArr = urls.slice(0, NUMBER_OF_REQUESTS).map(url => {
+    return makeRequest(url, counter++);
   });
 
   return Promise.all(promiseArr)
     .then(arr => {
-      const average = (arr[0] + arr[1]) / 2;
+      console.log('Arr all: ', arr);
```
Comment (ptroadec) [inline downloadSpeed.js]: to remove

File: downloadSpeed.js [hunk 2]
```diff
@@ -78,3 +28,78 @@ function checkDownloadSpeed(urls) {
...
+
+      response.once('data', () => {
+        console.log(`Url ${url} started receiving data`);
+        startTime = new Date().getTime();
+      });
+
+      // Not all servers (due to availability) have files of the same sizes, hence the need to standarize
+      response.on('data', chunk => {
+        if (url === 'http://ping.online.net/500Mo.dat') {
+          // console.log('500 MB url is receiving data');
```
Comment (ptroadec) [inline downloadSpeed.js]: is this used anymore ?

File: downloadSpeed.js [hunk 3]
```diff
@@ -78,3 +28,78 @@ function checkDownloadSpeed(urls) {
...
+          case 1:
+            if (Buffer.concat(arr).length <= thresholds[1] * 1_000_000) {
+              arr.push(chunk);
+              return;
+            }
+            break;
+          case 2: {
+            if (Buffer.concat(arr).length <= thresholds[2] * 1_000_000) {
+              // 3 000 000
+              // console.log('500 MB threshold not passed');
```
Comment (ptroadec) [inline downloadSpeed.js]: to remove

File: downloadSpeed.js [hunk 4]
```diff
@@ -78,3 +28,78 @@ function checkDownloadSpeed(urls) {
 }
 
 module.exports = checkDownloadSpeed;
```
Comment (ptroadec) [inline downloadSpeed.js]: put this line at the end of file

File: downloadSpeed.js [hunk 5]
```diff
@@ -78,3 +28,78 @@ function checkDownloadSpeed(urls) {
...
+              // console.log('500 MB threshold not passed');
+              arr.push(chunk);
+              return;
+            }
+            console.log('100 MB threshold PASSED!');
+            const thresholdIntervalId = setInterval(() => {
+              return console.log('100 MB threshold PASSED!');
+            }, 1500);
+
+            setTimeout(() => clearInterval(thresholdIntervalId), 15_000);
```
Comment (ptroadec) [inline downloadSpeed.js]: this may prevent CLI termination until printed

File: downloadSpeed.js [hunk 6]
```diff
@@ -78,3 +28,78 @@ function checkDownloadSpeed(urls) {
...
+      });
+
+      // Not all servers (due to availability) have files of the same sizes, hence the need to standarize
+      response.on('data', chunk => {
+        if (url === 'http://ping.online.net/500Mo.dat') {
+          // console.log('500 MB url is receiving data');
+        }
+        switch (counter) {
+          case 0:
+            if (Buffer.concat(arr).length <= thresholds[0] * 1_000_000) {
```
Comment (ptroadec) [inline downloadSpeed.js:50]: calling Buffer.concat at each data event is costly in terms of performance

File: downloadSpeed.js [hunk 7]
```diff
@@ -1,75 +1,25 @@
...
-            const kbps = (bps / 1024).toFixed(2);
-            const mbps = +(kbps / 1024).toFixed(2);
-
-            resolve(mbps);
-          });
-        });
-      }
-    });
+  promiseArr = urls.slice(0, NUMBER_OF_REQUESTS).map(url => {
+    return makeRequest(url, counter++);
```
Comment (ptroadec) [inline downloadSpeed.js]: index of url in array can be used instead of counter variable

File: downloadSpeed.js (hunk 2 above)
Comment (FarisMarouane) [reply downloadSpeed.js]: no

**Review by ptroadec — APPROVED** [review]

**Pull request merged**